azure/scope/cluster.go

```diff
@@ -128,12 +131,24 @@ func (s *ClusterScope) PublicIPSpecs() []azure.PublicIPSpec {
 		}
 	}
 
-	return []azure.PublicIPSpec{
-		controlPlaneOutboundIP,
-		{
+	publicIPSpecs = append(publicIPSpecs, controlPlaneOutboundIP)
+
+	// Node outbound public ip spec
+	// If no value is configured for LoadBalancerNodeOutboundIPs, or if it is set to 1, create one public ip
+	if s.LoadBalancerNodeOutboundIPs() == nil || *s.LoadBalancerNodeOutboundIPs() == 1 {
+		return append(publicIPSpecs, azure.PublicIPSpec{
 			Name: azure.GenerateNodeOutboundIPName(s.ClusterName()),
-		},
+		})
 	}
+
+	// For LoadBalancerNodeOutboundIPs > 1 create as many public ips as configured in LoadBalancerNodeOutboundIPs
```

Comment:
If we update `LoadBalancerNodeOutboundIPs()` to return `1` if the underlying data is `nil` (which I assume means the user did not provide a value, and we would provide a default value of `1`) then we could reduce some code surface area here and simply use the enumeration for all values of `LoadBalancerNodeOutboundIPs()`.

Reply:
The reason I've handled `LoadBalancerNodeOutboundIPs == 1` logic separately is I because I wanted the naming of azure resources to be the same as it was before (without the `-1` suffix). It's a bit of redundant code but I didn't want to behavior to change when there's no value set, which I believe will be, in most cases. Does that make sense?

Reply:
Makes sense.

How about we capture the value as a variable so we only have to call `s.LoadBalancerNodeOutboundIPs()` only once.

And just a suggestion: IMO formulating these two append patterns (either == 1 or > 1) as an if/else will have the effect of reading the code more suggestive that these are mutually exclusive; as opposed to the early return statement in the `== 1` scenario. Keeping the common `return publicIPSpecs` at the end of the function as the only return invocation is a little easier to read/maintain.

Reply:
Thanks for the feedback. I've made the changes, ptal.